### DISCUSSION TITLE
Feedback for...everything

### DISCUSSION THREAD

**Discussion by steveiliop56** [TimWitzdam/GitSave#3] [General]
Hello!

Awesome project! I just have some feedback to help you improve. Firstly you need a better readme, be more descriptive, include screenshots (I like your animation but I need a screenshot too), a docker compose, license information and contribution guides so people can see your project in action, for example I would love to see a gif or a video demonstrating the project. Secondly a lot of people are migrating docker images to GitHub because you can integrate it with GitHub actions and to be honest I like seeing an image that I run on my system being published automatically rather than a person manually building and uploading it. Furthermore I would strongly advise you to use semver (v`major`.`feature`.`patch`) so your versions are easily understandable and automatically picked up by bots like renovate. Last but not least when looking the screenshots I didn't see a way to clone this repository meaning that I have a backup...I can't use. It would be nice to have an option to clone the repository since it would make the use of backups easier. That's it from me, keep up with the good work!

**Comment by nhalstead** [comment]
Very similar thoughts, just broke it down to a bullet list.

- More descriptive Readme (include screenshots in real world settings)
  - Functional examples, diagrams, compatibility charts
- a docker compose template
  - Nothing extravigant, simple copy paste, run example like this: [Wizarr Getting Started](https://github.com/wizarrrr/wizarr/tree/4a7b4737fd4ff37d5e1f0c99b12c66ffecb9867c?tab=readme-ov-file#getting-started)
- license information and contribution guides so people can see your project in action
  - IE: a gif or a video demonstrating the project
  - You already have a license, just mention the kind of license in the Readme.
- Github Action for building the docker image
  - This starts the line of trust, seeing how the sausage is made
- Github Container Registry for the image
  - This is another ring is the line of trust
  - We can see the [dockerhub](https://hub.docker.com/r/timwitzdam/gitsave/tags) registry, but it's easier to have it all in one place as an audit.
- Follow semver (vmajor.feature.patch)
   - Keeps things organized. For docker images you can tag them with the version (very nice to have). For an easy start, tag them with the latest tag, **and** the short git hash (easy to obtain).
- Instructions to clone the repo
  - Update the [.gitattributes](https://git-scm.com/docs/gitattributes#_export_ignore) file to exclude irrelevant files for an archive
  - This is an easy win overall, and can fall under the Readme updates.

**Comment by TimWitzdam** [comment]
First at all, thank you guys for so many ideas haha
There's so much I would have never actually thought about but make total sense. I'll need to read a little bit of documentation regarding GitHub actions/Container Registry, but I'll definitely add those changes in the future.
